src/default-client/index.ts

```diff
@@ -0,0 +1,840 @@
+    const analyzeResult = await this._cache.analyze(queryHash, ast, context.fieldTypeMap);
+    const { cachedData, cacheMetadata, filtered, updatedAST, updatedQuery } = analyzeResult;
+    const deferred: DeferPromise.Deferred<void> = deferredPromise();
+
+    if (cachedData && cacheMetadata) {
+      const defaultCacheControl = this._defaultCacheControls.query;
+      const queryCacheability = cacheMetadata.get("query");
+      const cacheControl = queryCacheability && queryCacheability.printCacheControl() || defaultCacheControl;
+
+      (async () => {
+        try {
+          await this._cache.responses.set(
+            queryHash,
+            { cacheMetadata: mapToObject(cacheMetadata), data: cachedData },
+            { cacheHeaders: { cacheControl }, tag: opts.tag },
+          );
+        } catch (error) {
+          // no catch
+        } finally {
+          deferred.resolve();
+        }
+      })();
+
+      return this._resolve({
+        cacheMetadata,
+        cachePromise: deferred.promise,
+        data: cachedData,
+        operation: "query",
+        queryHash,
+      });
+    }
+
+    let fetchResult: FetchResult;
+    const _updateQuery = updatedQuery as string;
+    const _updateAST = updatedAST as DocumentNode;
+    const _filterd = filtered as boolean;
+
+    try {
+      fetchResult = await this._fetch(_updateQuery);
+    } catch (error) {
+      return this._resolve({ error, operation: "query", queryHash });
+    }
+
+    const _cacheMetadata = createCacheMetadata({
+      cacheMetadata: fetchResult.cacheMetadata,
+      headers: fetchResult.headers,
+    });
+
+    const resolveResult = await this._cache.resolveQuery(
+      _updateQuery,
+      _updateAST,
+      queryHash,
+      context.fieldTypeMap,
+      fetchResult.data,
+      _cacheMetadata,
+      { cacheResolve: deferred.resolve, filtered: _filterd, tag: opts.tag },
+    );
+
+    return this._resolve({
+      cacheMetadata: resolveResult.cacheMetadata,
+      cachePromise: deferred.promise,
+      data: resolveResult.data,
+      operation: "query",
+      queryHash,
+    });
+  }
+
+  private async _resolve(args: ResolveArgs): Promise<ResolveResult> {
+    const { cacheMetadata, cachePromise, data, error, operation, queryHash, resolvePending = true } = args;
+
+    if (cacheMetadata && !cacheMetadata.has("query")) {
+      const cacheability = new Cacheability();
+      cacheability.parseCacheControl(this._defaultCacheControls[operation]);
+      cacheMetadata.set("query", cacheability);
+    }
+
+    if (resolvePending && queryHash) {
+      this._resolvePendingRequests(queryHash, cacheMetadata, data, error);
+      this._cache.requests.active.delete(queryHash);
+    }
+
+    if (error) return Promise.reject(error);
+
+    const output: ResolveResult = {
+      cacheMetadata,
+      data: data as ObjectMap,
+    };
+
+    if (cachePromise) output.cachePromise = cachePromise;
+    if (queryHash) output.queryHash = queryHash as string;

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4325)